opentelemetry-resourcedetector-gcp/src/opentelemetry/resourcedetector/gcp_resource_detector/_mapping.py

```diff
@@ -186,10 +195,19 @@ def _create_monitored_resource(
     for mr_key, map_config in mapping.items():
         mr_value = None
         for otel_key in map_config.otel_keys:
-            if otel_key in resource_attrs:
+            if otel_key in resource_attrs and not str(
+                resource_attrs[otel_key]
+            ).startswith(_constants.UNKNOWN_SERVICE_PREFIX):
                 mr_value = resource_attrs[otel_key]
                 break
 
+        if (
```

Comment:
Won't this overwrite any MR label with service.name if it's None? For example location won't become "global" if there is a service.name set

Reply:
The next line, `and ResourceAttributes.SERVICE_NAME in map_config.otel_keys`, should mean that it only overwrites MR labels with service.name if service.name is one of the keys that was one of the fallbacks for the MR label.